lib/galaxy/config/__init__.py

```diff
@@ -716,6 +717,11 @@ def sentry_dsn_public(self):
         else:
             return None
 
+    def is_set(self, property):
+        if property not in self._raw_config:
+            raise Exception("Property does not exist: '%s'" % property)
```

Comment:
There are some debugging things that are not in the schema ... personally I think logging a warning would be enough here, but not opposed to documenting everything either.

Reply:
Here's an example: it is used somewhere in code to check if some config file is set, and if no, fall back to the corresponding sample config. And the config file property name is misspelled in that place - so, as a result, it will always fall back to the sample, logging a warning (which will be lost in the output). Also, since this is used by developers and not by users or admins, this may happen only during development. So, I think, failing fast and loudly would be a safer option here? (i.e., "stop now! you are checking a non-existent config...")

As for the debugging use cases, I think they will simply use getattr(), as they do now. In any case, we certainly won't just remove all `foo_set` attributes without ensuring that any `foo` is either in the schema, or is recognized by the new `is_set(foo)` method (with some edit), or has another way to check if it was set.

Reply:
@mvdbeek may i mark this as resolved? Also, if you think a warning is still a better choice, I'm perfectly fine with that.